parser/src/main/java/com/microsoft/gctoolkit/parser/jvm/Decorators.java

```diff
@@ -38,31 +64,29 @@ private void extractValues(String line) {
         if (!line.startsWith("["))
             return;
```

Comment:
I wonder if it wouldn't be better to let this fall through. The matcher won't match, but at least it won't be null.

Reply:
was thinking the same but then opt'ed to do something else.. I would opt to replace this entire constructor with some form of a builder as the logic is too complex for a constructor (as John noted). But I'd to that in a separate pull request if you don't mind